go-controller/pkg/ovn/ovn.go

```diff
@@ -1055,6 +1063,43 @@ func (oc *Controller) GetServiceVIPToName(vip string, protocol kapi.Protocol) (t
 	return namespace, ok
 }
 
+// GetNamespacePolicyACLLogging retrieves ACL deny policy logging setting for the Namespace
+func (oc *Controller) GetNamespacePolicyACLLogging(ns string) *ACLLoggingLevels {
+	nsInfo := oc.getNamespaceLocked(ns)
+	if nsInfo == nil {
+		return &ACLLoggingLevels{
+			Allow: "",
+			Deny:  "",
+		}
+	}
+	defer nsInfo.Unlock()
+	return &nsInfo.aclLogging
+}
+
+// Verify if controller can support ACL logging and validate annotation
+func (oc *Controller) aclLoggingCanEnable(annotation string, nsInfo *namespaceInfo) bool {
+	if !oc.aclLoggingEnabled {
+		return false
+	}
+	var aclLevels ACLLoggingLevels
+	err := json.Unmarshal([]byte(annotation), &aclLevels)
+	if err != nil {
+		return false
+	}
+	okCnt := 0
+	for _, s := range []string{"alert", "warning", "notice", "info", "debug"} {
+		if aclLevels.Deny != "" && s == aclLevels.Deny {
+			nsInfo.aclLogging.Deny = aclLevels.Deny
+			okCnt++
+		}
+		if aclLevels.Allow != "" && s == aclLevels.Allow {
+			nsInfo.aclLogging.Allow = aclLevels.Allow
+			okCnt++
+		}
+	}
```

Comment:
I would expect any of the following to be valid:
```
1. k8s.ovn.org/acl-logging: '{ "deny": "alert", "allow": "notice" }'
2. k8s.ovn.org/acl-logging: '{ }'
3. k8s.ovn.org/acl-logging: '{ "deny": "alert"}'
4. k8s.ovn.org/acl-logging: '{ "allow": "alert"}'
```

(2) above should be valid since per the definition below they can be empty and also it is the easiest way to author the request to enable logging.
```
// ACL logging severity levels
type ACLLoggingLevels struct {
	Allow string `json:"allow,omitempty"`
	Deny  string `json:"deny,omitempty"`
}
```

Reply:
however, I wanted to provide a way to configure deny or allow only stanzas, i.e. cases (3) and (4). The case (2) is undefined, and I think it should not be used to enable default levels for both. It is still useful to allow 'omitempty' when administrator wants to temporarily disable both but keep annotations around..